### PR TITLE
Omtfdev mk15

### PR DESCRIPTION
**Pull request by konec**
OMTF: a small hack to hide GlobalTag dependency in emulator. Well improves data-emul agreement
@kbunkow @akalinow @konec

### CODE REVIEW

File: L1Trigger/L1TMuonOverlap/src/AngleConverter.cc [hunk 1]
```diff
@@ -88,6 +88,20 @@ int etaKeyWG2Code(const CSCDetId& detId, uint16_t keyWG) {
   return etaCode;
 }
 
+
+int fixCscOffsetGeom(int offsetLoc) {
+  // fix for CSC feo dependance from GlobalTag
+
+  // dump of CSC offsets for MC global tag
+  std::vector<int> offCSC = { -154, -133, -17, -4, 4, 17, 133, 146, 154, 167,
```
Comment (rekovic) [inline L1Trigger/L1TMuonOverlap/src/AngleConverter.cc:96]: @konec 
Please document where we get these magic numbers from,
or if possible what do these numbers correspond to.
Note: documentation of constant numbers is a requirement 
before merging code into CMSSW.

Comment (rekovic) [reply L1Trigger/L1TMuonOverlap/src/AngleConverter.cc:96]: @konec 
Will merge into l1t-integration branch, but please make sure
you make another PR with the comment above this line explaining
what these numbers represent and how they are obtained.  Thanks.

**Comment by fwyzard** [comment]
Did you get feedback on what the numbers mean, and where they come from ?

Also,what is the whole point of this PR ?
Does it mean there is a disagreement in the data/MC geometry ?

**Comment by rekovic** [comment]
The PR was merged and tagged as v96.24.  

It effects the re-emulation of OMTF when its inputs are unpacked csctfDigis.
In L1T re-emulation workflows, this was the case with 
L1TReEmulFromRAW2015 and L1TReEmulFromRAWCalouGT workflows. 
It does not concern emulation of OMTF in MC. 

Now regaring data:
In 2017, for part of the data there are now longer csctfDigis available, which 
calls for further profilferation of re-emulation workflows.  This is not desirable.
Instead, I am introducing a new modifier stage2Trigger_2017 which takes 
care of this internally to customisation functions. Will publish a new tag soon and provide explanation/twiki.

**Comment by rekovic** [comment]
Will address the L1REPACK as well.

**Comment by rekovic** [comment]
Correction:

`L1TReEmulFromRAWCalouGT` workflow is not affected by this PR,
as that workflow uses unpacked uGMT muons as input to uGT.
`L1TReEmulFromRAW2015` workflow is the only one affected.
This comment refers to the final decisions of uGT.

Special comment to the users of L1T Ntuples:
If using `L1TReEmulFromRAWCalouGT` in combination with filling emulated L1TNtuples 
```--customise=L1Trigger/L1TNtuples/customiseL1Ntuple.L1NtupleEMU```, the emulated uGMT
muons from OMTF will be affected.

**Comment by rekovic** [comment]
Updated comment above.

**Comment by fwyzard** [comment]
Hi Vladimir,
sure, I think I agree on all points - but why does this need a code change, and not only a change in configuration or conditions ?

.A

**Comment by rekovic** [comment]
@fwyzard 
I agree. These sorts of things should indeed be provided as parameters in configurations or conditions,
preferable latter. 

@konec 
Could you move these constants to OMTF conditions in the next PR, please.

**Comment by rekovic** [comment]
@konec 
Any news here?

**Comment by rekovic** [comment]
@konec 
Can you also move the constants in
EventFilter/L1TRawToDigi/data/OmtfRpcLinksMap.txt to a configuration or preferably to conditions.

**Comment by konec** [comment]
@rekovic 
Hi, The numbers in question are offset parameters which are applied for computation of internal phi.
Up to now they were computed from Global Tag (GT). In hardware pairs of values specifying link number and offsets corresponding to MonteCarlo GT are used. For cmssw they are recalculated. However, since  for data processing different GT is used the computed values are substituted with those from MonteCarlo GT . In fact it is commented "dump of CSC offsets for MC global tag".
That is the simplest fix I found to fix data-emulatror disagreement.

**Pull request merged**